src/python/pants/bsp/util_rules/targets.py

```diff
@@ -64,23 +129,160 @@ class WorkspaceBuildTargetsHandlerMapping(BSPHandlerMapping):
     response_type = WorkspaceBuildTargetsResult
 
 
+@dataclass(frozen=True)
+class GenerateOneBSPBuildTargetRequest:
+    bsp_target: BSPBuildTargetInternal
+
+
+@dataclass(frozen=True)
+class GenerateOneBSPBuildTargetResult:
+    build_target: BuildTarget
+    digest: Digest = EMPTY_DIGEST
+
+
+def find_metadata_merge_order(
+    metadata_request_types: Sequence[Type[BSPBuildTargetsMetadataRequest]],
+) -> Sequence[Type[BSPBuildTargetsMetadataRequest]]:
+    if len(metadata_request_types) <= 1:
+        return metadata_request_types
+
+    # Naive algorithm (since we only support Java and Scala backends), find the metadata request type that cannot
+    # merge from another and put that first.
```

Comment:
I don't really follow what this code is doing. Sounds like this will be generalized in some future? I was going to ask a unit test but probably not worth it here?